sourcebundle/builder.go

```diff
@@ -713,3 +733,20 @@ func packagePrepareWalkFn(root string, ignoreRules *ignorefiles.Ruleset) filepat
 		return nil
 	}
 }
+
+func extractVersionListFromResponse(modPackageInfos []ModulePackageInfo) versions.List {
+	vs := make(versions.List, len(modPackageInfos))
+	for _, v := range modPackageInfos {
+		vs = append(vs, v.Version)
+	}
```

Comment:
Ah that makes perfect sense, thanks for the go tip! 

From what I can surmise it seems that either approach doesn't have a specific benefit over the other in this instance. Since we know that in the end the length of the slice will always be equal to its capacity. So allocating memory for all the element up front, as in the first suggestion, vs doing so within the loop with append in total doesn't make a difference. Is my thinking correct?

Reply:
The only difference it makes is that allocating the space up front saves time and memory because it doesn't require that the slice grow after some appends happen. When a slice grows, it can allocate not enough or too much as well as copy all the elements from the old slice to the new, larger one. If you know the size slice you need, it's usually better to use the first option.

Reply:
Gotcha, thanks for the more detailed explanation. Included 👍 https://github.com/hashicorp/go-slug/pull/59/commits/3fa65750f5638397e8c92aa9bc469e4d7d628c52